### PR TITLE
Chat channel switching improvements

### PR DESCRIPTION
**Pull request by Clyybber**
## About the PR 

Default to IC channels when joining, switch to another IC channel if the current one disappears before trying OOC.
Allow chat channel cycling while the textbox is focused.

**Changelog**

:cl:
- remove: Chat no longer defaults to OOC if IC channels are available.
- add: Can now cycle channels while typing.
- fix: Focus channel keybinds now actually focus the chatbox.

### CODE REVIEW

File: Content.Shared/Chat/ChatChannel.cs [hunk 1]
```diff
@@ -61,5 +61,10 @@ public enum ChatChannel : short
         ///     Unspecified.
         /// </summary>
         Unspecified = 512,
+
+        /// <summary>
+        ///     Channels considered to be IC.
+        /// </summary>
+        IC = Local | Radio | Damage | Radio | Visual | Emotes | Dead,
```
Comment (mirrorcult) [inline Content.Shared/Chat/ChatChannel.cs:68]: radio is put twice

Comment (Clyybber) [reply Content.Shared/Chat/ChatChannel.cs:68]: oh, thx

**Review by ShadowCommander — APPROVED** [review]
Other than this and what mirrorcult put then it looks good.

File: Content.Client/Chat/ChatBox.cs [hunk 2]
```diff
@@ -498,6 +504,25 @@ protected override void KeyBindUp(GUIBoundKeyEventArgs args)
...
 
+        public void CycleChatChannel(bool forward)
+        {
+            Input.IgnoreNext = true;
+            var channels = SelectableChannels;
+            var idx = channels.IndexOf(SelectedChannel);
+            if (forward)
+            {
+                idx++;
+                idx = MathHelper.Mod(idx, channels.Count());
```
[the remark this answers is not in the document]
Comment (Clyybber) [reply Content.Client/Chat/ChatBox.cs:515]: True, I just moved it from GameScreen.cs to here

**Pull request merged**